tests/integration/test_charm.py

```diff
@@ -31,6 +31,11 @@ async def test_deploy_active(ops_test: OpsTest):
     assert ops_test.model.applications[APP_NAME].status == "active"
 
 
+@pytest.mark.abort_on_fail
+async def test_consistency_between_workload_and_metadata(ops_test: OpsTest):
+    application = ops_test.model.applications[APP_NAME]
+    assert application.data.get("workload-version", "") == DEPENDENCIES["service"]["version"]
```

Comment:
While doing this, I noticed a small mismatch with `kafka` where the key is called `kafka-service`. It would be good to keep these consistent IMHO, but I would leave it up to you to make a call on this

Reply:
I recall there are other small discrepancies on `literals` and `helpers` generally. It should be fine to address them as we go.

Reply:
Actually, I believe it makes then more sense to have "service" in both. Anyhow, I would honestly try to scout all these differences and bundle some modifications together (also deciding whether to go for A or B)